src/ServiceControl.AcceptanceTests/Audit/When_an_error_message_fails_to_import.cs

```diff
@@ -0,0 +1,137 @@
+﻿namespace ServiceBus.Management.AcceptanceTests.Audit
+{
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Threading.Tasks;
+    using EndpointTemplates;
+    using NServiceBus;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTesting.Customization;
+    using NServiceBus.AcceptanceTests;
+    using NServiceBus.Faults;
+    using NServiceBus.Settings;
+    using NUnit.Framework;
+    using ServiceControl.Infrastructure;
+    using ServiceControl.Operations;
+
+    /// <summary>
+    /// This is a specific test https://github.com/Particular/PlatformDevelopment/issues/2464 - an error message that ends up in failed audits collection
+    /// To simulate this we used audit message that contains an artificial FailedQ header
+    /// </summary>
+    class When_an_error_message_fails_to_import : AcceptanceTest
+    {
+        [Test]
+        public async Task It_can_be_reimported()
+        {
+            //Make sure the audit import attempt fails
+            CustomConfiguration = config => { config.RegisterComponents(c => c.ConfigureComponent<FailOnceEnricher>(DependencyLifecycle.SingleInstance)); };
+
+            FailedAuditsCountReponse failedImportsCountReponse;
+
+            await Define<MyContext>()
+                .WithEndpoint<Sender>(b => b.When((bus, c) =>
+                {
+                    var ops = new SendOptions();
+                    ops.SetHeader(FaultsHeaderKeys.FailedQ, "SomeFailedQ");
+                    return bus.Send(new MyMessage(), ops);
+                }))
+                .WithEndpoint<Receiver>()
+                .Done(async c =>
+                {
+                    if (c.UniqueMessageId == null)
+                    {
+                        return false;
+                    }
+
+                    if (!c.WasReimported)
+                    {
+                        var countResult = await this.TryGet<FailedAuditsCountReponse>("/api/failedaudits/count");
```

Comment:
I'm wondering if it would be possible to make this test step ID based. As long as we create a new instance of SC with a temporary folder the count based approach will work but I think the ID based one, in this case, would be more robust.

Reply:
I looked at it but it would be hard because the failed audit/error documents have auto-generated IDs so accessing them by ID would require adding an index :\